File: app/src/main/java/com/theone/simpleshare/ui/paired/RecyclerAdapter.java
package com.theone.simpleshare.ui.paired;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.theone.simpleshare.R;

import java.util.ArrayList;


public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ViewHolder> {


    private ArrayList<PairedItem> mItemList;
    Context mContext;
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.paired_item_recyclerview, parent, false);
        mContext = parent.getContext();
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.onBind(mItemList.get(position));
    }
    @Override
    public int getItemCount() {
        return mItemList.size();
    }

    public void setItemList(ArrayList<PairedItem> list){
        this.mItemList = list;
        notifyDataSetChanged();
    }
    public PairedItem getItemFromList(int position){
        return this.mItemList.get(position);
    }

    public void removeItemFromList(int position){
        this.mItemList.remove(position);
        notifyItemRemoved(position);
        notifyItemRangeChanged(position, getItemCount());
    }

    public interface OnItemClickListener {
        void onItemClick(View v, int pos , PairedItem item);
    }
    private  OnItemClickListener mListener=null;
    public void setOnItemClickListener( OnItemClickListener listener){
        this.mListener = listener;
    }

    public void clear(){
        mItemList.clear();
        notifyDataSetChanged();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageView profile;
        TextView name, address, state, batteryLevel;
        public ViewHolder(@NonNull View itemView){
            super(itemView);

            profile = itemView.findViewById(R.id.profile);
            name = itemView.findViewById(R.id.name);
            address = itemView.findViewById(R.id.address);
            state = itemView.findViewById(R.id.state);
            batteryLevel = itemView.findViewById(R.id.battery_level);

            itemView.setOnClickListener( new View.OnClickListener(){
                @Override
                public void onClick(View view) {
                    int pos = getAdapterPosition();
                    if( pos != RecyclerView.NO_POSITION){
                        mListener.onItemClick(view, pos , mItemList.get(pos) );
                    }
                }
            });

        }

        void onBind(PairedItem item){
            profile.setImageResource(item.getResourceId());
            name.setText(item.getName());
            address.setText(item.getAddress());
            state.setText(""+ item.state);
            batteryLevel.setText(item.batteryLevel + "%");

        }

    }

}

File: app/src/main/kotlin/com/theone/simpleshare/ui/paired/RecyclerAdapter.kt
package com.theone.simpleshare.ui.paired

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.theone.simpleshare.R
import java.util.ArrayList

class RecyclerAdapter : RecyclerView.Adapter<RecyclerAdapter.ViewHolder?>() {
    private lateinit var mItemList: ArrayList<PairedItem>
    private lateinit var mListener: OnItemClickListener
    private lateinit var mContext: Context
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.paired_item_recyclerview, parent, false)
        mContext = parent.getContext()
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.onBind(mItemList!![position])
    }

    fun setItemList(list: ArrayList<PairedItem>?) {
        if (list != null) {
            mItemList = list
        }
        notifyDataSetChanged()
    }

    fun getItemFromList(position: Int): PairedItem = mItemList[position]

    fun removeItemFromList(position: Int) {
        mItemList!!.removeAt(position)
        notifyItemRemoved(position)
        notifyItemRangeChanged(position, itemCount)
    }

    interface OnItemClickListener {
        fun onItemClick(v: View?, pos: Int, item: PairedItem)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        mListener = listener
    }

    fun clear() {
        mItemList.clear()
        notifyDataSetChanged()
    }

    open inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var profile: ImageView
        var name: TextView
        var address: TextView
        var state: TextView
        var batteryLevel: TextView
        fun onBind(item: PairedItem) {
            profile.setImageResource(item.resourceId)
            name.text=item.name
            address.text=item.address
            state.text = "" + item.state
            batteryLevel.text = item.batteryLevel.toString() + "%"
        }

        init {
            profile = itemView.findViewById(R.id.profile)
            name = itemView.findViewById<TextView>(R.id.name)
            address = itemView.findViewById<TextView>(R.id.address)
            state = itemView.findViewById<TextView>(R.id.state)
            batteryLevel = itemView.findViewById<TextView>(R.id.battery_level)
            itemView.setOnClickListener { view ->
                val pos: Int = getAdapterPosition()
                if (pos != RecyclerView.NO_POSITION) {
                    mListener.onItemClick(view, pos, mItemList[pos])
                }
            }
        }
    }

    override fun getItemCount(): Int = mItemList.size


}